provider/cloudflare/cloudflare.go

```diff
@@ -575,8 +550,8 @@ func (p *CloudFlareProvider) submitDataLocalizationRegionalHostnameChanges(ctx c
 // dataLocalizationRegionalHostnamesChanges processes a slice of cloudFlare changes and consolidates them
 // into a list of data localization regional hostname changes.
 // returns nil if no changes are needed
-func dataLocalizationRegionalHostnamesChanges(changes []*cloudFlareChange) ([]DataLocalizationRegionalHostnameChange, error) {
-	regionalHostnameChanges := make(map[string]DataLocalizationRegionalHostnameChange)
+func dataLocalizationRegionalHostnamesChanges(changes []*cloudFlareChange) ([]regionalHostnameChange, error) {
```

Comment:
^^ (would it make sense to move this to `cloudflare_regional.go` ?)

Reply:
My intention was to move only some parts of the regional services logic, since my other PR ([#5309](https://github.com/kubernetes-sigs/external-dns/pull/5309)) will handle the rest.
However, you’re right. It ultimately belongs in `cloudflare_regional.go`.
I can move it in this PR if you prefer.

Reply:
I moved it to `cloudflare_regional.go`